### PR TITLE
114 rebase project architecture

### PR DESCRIPTION
**Pull request by LucasTesnier**
Change all the library from Archive to SHARED. (Communicator, ECS, and Transisthor stayed in Archive for the moment (Cyclic dependces. Will be fixed in part two).

Polish all libraries link and reduce compilation time.

Clean Cmake of testing.

### CODE REVIEW

File: tests/unit_tests/libs_tests/ECS_tests/Entity_tests/CMakeLists.txt [hunk 1]
```diff
@@ -1,19 +1,11 @@
-
-## Setting Variables ##
-set (INCLUDE_ROOT ${PROJECT_SOURCE_DIR}/tests/unit_tests/libs_tests/ECS_tests/Entity_tests)
 set (SRC_ROOT ${PROJECT_SOURCE_DIR}/tests/unit_tests/libs_tests/ECS_tests/Entity_tests)
 
 set(TEST_SRC_ROOT
     ${SRC_ROOT}/Entity_class_tests.cpp
 )
 
-## Documented Example ##
```
Comment (31Nathan) [inline tests/unit_tests/libs_tests/ECS_tests/Entity_tests/CMakeLists.txt:10]: Do NOT Remove Cmake documentation

File: tests/unit_tests/libs_tests/SFML_tests/CMakeLists.txt [hunk 2]
```diff
@@ -13,13 +10,8 @@ set(TEST_SRC_ROOT
     ${SRC_ROOT}/GraphicsTextComponentTests.cpp
 )
 
-## Documented Example ##
 create_test(
-    ## First, the name of the test ##
```
Comment (31Nathan) [inline tests/unit_tests/libs_tests/SFML_tests/CMakeLists.txt:18]: And Last

**Review by 31Nathan — CHANGES_REQUESTED** [review]

**Review by ZQUEMA — APPROVED** [review]

File: tests/unit_tests/libs_tests/ECS_tests/Entity_tests/CMakeLists.txt (hunk 1 above)
Comment (LucasTesnier) [reply tests/unit_tests/libs_tests/ECS_tests/Entity_tests/CMakeLists.txt:10]: IT's the template documentation. She has nothing to do here

File: tests/unit_tests/libs_tests/ECS_tests/World_tests/CMakeLists.txt [hunk 3]
```diff
@@ -1,19 +1,12 @@
 
-## Setting Variables ##
-set (INCLUDE_ROOT ${PROJECT_SOURCE_DIR}/tests/unit_tests/libs_tests/ECS_tests/World_tests)
 set (SRC_ROOT ${PROJECT_SOURCE_DIR}/tests/unit_tests/libs_tests/ECS_tests/World_tests)
 
 set(TEST_SRC_ROOT
     ${SRC_ROOT}/World_class_tests.cpp
 )
 
-## Documented Example ##
```
[the remark this answers is not in the document]
Comment (LucasTesnier) [reply tests/unit_tests/libs_tests/ECS_tests/World_tests/CMakeLists.txt:10]: IT's the template documentation. She has nothing to do here

File: tests/unit_tests/libs_tests/SFML_tests/CMakeLists.txt (hunk 2 above)
Comment (LucasTesnier) [reply tests/unit_tests/libs_tests/SFML_tests/CMakeLists.txt:18]: IT's the template documentation. She has nothing to do here

**Review by 31Nathan — APPROVED** [review]
LGTM!

**Pull request merged**